esmvalcore/preprocessor/_regrid.py

```diff
@@ -1288,8 +1308,10 @@ def extract_levels(
         result = cube
         # Set the levels to the requested values
         src_levels.points = levels
-    elif len(src_levels.shape) == 1 and set(levels).issubset(
-        set(src_levels.points)
+    elif len(src_levels.shape) == 1 and (
+        set(levels).issubset(set(src_levels.points))
+        if len(levels.shape) == 1
+        else False
```

Comment:
```suggestion
    elif len(src_levels.shape) == 1 and 
        len(levels.shape) == 1 and
        set(levels).issubset(set(src_levels.points))
```

probably easier to read.

Reply:
I wrote it like that because in the case of `levels` being an `np.array` of dimension > 1, the `set(levels)` call will return an error (`TypeError: unhashable type: 'numpy.ndarray'`). The inline `if ... else` statement allowed to prevent that.
However, to keep it clean I will commit your suggestion but change it to `set(levels.flatten())` to avoid any dimension problem.

Reply:
Actually that should work without the `flatten`. If one of the statements evaluates to False in this `and` cascade, the remaining ones are not checked. Thus, the `set(levels).issubset(set(src_levels.points))` will only be run if `len(levels.shape) == 1` is `True`. But that doesn't really matter, you can leave it like that!

Reply:
I see, I only had the issue because of the previous order of the statements then.